include/CXXGraph/Graph/Algorithm/Dinic_impl.hpp

```diff
@@ -0,0 +1,87 @@
+#include <iostream>
+#include <queue>
+#include <limits>
+#include <unordered_map>
+#include <vector>
+#include "Node_impl.h"
+#include "CXXGraph/Graph/Graph_decl.h"
+
+
+namespace CXXGraph {
+
+template <typename T>
+class Graph {
+private:
+    std::unordered_map<CXXGraph::id_t, Node<T>> nodes;
+    std::unordered_map<CXXGraph::id_t, std::unordered_map<CXXGraph::id_t, int>> capacities;
+    std::unordered_map<CXXGraph::id_t, std::unordered_map<CXXGraph::id_t, int>> flow;
```

Comment:
Also, why is he re-declaring the Graph class? Shouldn't he simply add the declaration of the method in `Graph_decl.hpp` and add a header for the implementation of the single method?

Reply:
I Agree!